daemon/fmutex/fmutex.go

```diff
@@ -0,0 +1,81 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package fmutex implements locking for the snappy daemon
+package fmutex
+
+import (
+	"sync"
+
+	"github.com/ubuntu-core/snappy/dirs"
+	"github.com/ubuntu-core/snappy/logger"
+	"github.com/ubuntu-core/snappy/priv"
+)
+
+// FMutex gives you an exclusive mutex that also holds the filesystem lock.
+type FMutex struct {
+	mutex sync.Locker
+	flock FLocker
```

Comment:
And just to be clear, it's not that I don't care about tests, but rather than I'd prefer to have the right implementation than to have a suboptimal one and comfort-inducing tests that do not in fact touch the relevant complexity of the implementation.